"""
6.	В программе генерируется случайное целое число от 0 до 100.
Пользователь должен его отгадать не более чем за 10 попыток. После каждой
неудачной попытки должно сообщаться больше или меньше введенное пользователем
число, чем то, что загадано. Если за 10 попыток число не отгадано,
то вывести загаданное число.

Решите через рекурсию. Решение через цикл не принимается.
Для оценки Отлично в этом блоке необходимо выполнить 5 заданий из 7
Для оценки Отлично в этом блоке необходимо выполнить 5 заданий из 7

Подсказка:
Базовый случай здесь - угадали число или закончились попытки
"""

import random as r


def guess(n=10, a=r.randint(0, 100)):
    z = int(input('Угадайте число от 1 до 100: '))
    if n > 1:
        if a > z:
            print(f"Загаданное число больше введенного.") + guess(n - 1)
        elif a < z:
            print(f"Загаданное число меньше введенного.") + guess(n - 1)
        else:
            print(f'Вы угадали.')
    else:
        print(f"Вы проиграли. Было загадано число - {a}")


print(guess())

# Не совсем понимаю почему после того как угадывается число
# выводится тайп эррор unsupported operand type(s) for +: 'NoneType' and 'NoneType'
